dev/tools/update_icons.dart

```diff
@@ -352,65 +362,57 @@ class _Icon {
 
     if (id.endsWith('_outlined') && id!='insert_chart_outlined') {
```

Comment:
I'm really confused about the insert_chart icons. The complete list today seems to be:

```
insert_chart e7dc
insert_chart_outlined e248
insert_chart_outlined_outlined e249
insert_chart_outlined_rounded f278
insert_chart_outlined_sharp ed4a
insert_chart_rounded f279
insert_chart_sharp ed4b
```

Are those two sets of icons? One set of icons with many variants? Seven icons?

Reply:
It looks like all of this is actually only done so that we can do the ios/android specialization. Maybe there's a better way to do this that allows us to totally ignore the icon type? Just look for icons that startsWith() the icon types we want to specialise?

Reply:
Yeah, it's two sets `insert_chart` + 3 variants (top), and `insert_chart_outlined` + 2 variants (bottom). 

<img width="330" alt="Screenshot 2021-02-26 at 09 06 00" src="https://user-images.githubusercontent.com/6655696/109273085-04310d00-7812-11eb-9453-e0751beaddb7.png">

I think it's still needed, at the very least, for the css class to display the correct icon in dartdoc. Otherwise, yeah doing this in a style agnostic way would be simpler

Reply:
Can we maybe just have the bit where we decide how to style it for CSS look to see if there's a trailing suffix and apply the style appropriately there, rather than actually tracking the style across the whole data model and so on? Seems like this would save us a lot of code and get rid of the bug we're dealing with here at the same time.

Reply:
> Is there a difference between "insert_chart_outlined e248" and "insert_chart_outlined_outlined e249"?

Not that I can tell

Reply:
I get what you're saying but I think you end up with messier code because there are ~~3~~ 2 places that depends on the style-related logic
1. Dartdoc `style.htmlSuffix`
2. Determining the `shortId` (e.g. 5g)
3. ~~Determining the`fullFlutterId` (five_g, five_g_outlined, five_g_rounded, five_g_sharp)~~ I simplified and removed this. 

So, cleaner still to figure out the style in the constructor.

Reply:
As I reread this I realized I was wrong and could simplify again :) I reordered `_Icon` to make it more readable. Lmk if this LGTY